common/test/acceptance/pages/studio/course_info.py

```diff
@@ -1,16 +1,150 @@
 from common.test.acceptance.pages.studio.course_page import CoursePage
+from common.test.acceptance.pages.studio.utils import type_in_codemirror, set_input_value
 
 
 class CourseUpdatesPage(CoursePage):
     """
     Course Updates page.
     """
-
     url_path = "course_info"
 
     def is_browser_on_page(self):
-        return self.q(css='body.view-updates').present
+        """
+        Returns whether or not the browser on the page and has loaded the required content
+
+        Check for the presence of handouts-content, when it is present the render function has completed
+        loading the updates and handout sections
+        """
+        return self.q(css='.handouts-content').present
+
+    def are_course_updates_on_page(self):
+        """
+        Is the course updates content Div loaded on the page
+        """
+        return self.q(css='article#course-update-view.course-updates').present
+
+    def is_new_update_button_present(self):
+        """
+        Checks for the presence of the new update post button
+        """
+        return self.q(css='.new-update-button').present
+
+    def click_new_update_button(self):
+        """
+        Wait for the handouts-content, otherwise the render of the page will clear the new update
+        form when loading content
+        """
+        click_css(self, '.new-update-button', require_notification=False)
+
+    def submit_update(self, message):
+        """
+        Adds update text to the new update CodeMirror form and submits that text
+
+        Arguments:
+            message (str): The message to be added and saved
+        """
+        type_in_codemirror(self, 0, message)
+        self.click_new_update_save_button()
+
+    def set_date(self, date):
+        """
+        Sets the updates date input to the provided value
+
+        Arguments:
+            date (str): Date string in the format DD/MM/YYYY
+        """
+        set_input_value(self, 'input.date', date)
+
+    def is_update_date(self, search_date):
+        """
+        Checks to see if the search date is present
+
+        Arguments:
+            search_date (str): e.g. 06/01/2013 would be found with June 1, 2013
+        """
+        date = self.q(css='.date-display').html
+        if len(date) > 0:
+            if search_date == date[0]:
+                return True
+        return False
+
+    def is_new_update_save_button_present(self):
+        """
+        Checks to see if the CodeMirror Update save button is present
+        """
+        return self.q(css='.save-button').present
+
+    def click_new_update_save_button(self):
+        """
+        Clicks the CodeMirror Update save button
+        """
+        click_css(self, '.save-button')
+
+    def is_edit_button_present(self):
+        """
+        Checks to see if the edit update post buttons if present
+        """
+        return self.q(css='div.post-preview .edit-button').present
```

Comment:
Nit: in general we use class names and don't specify the HTML element. For instance,

return self.q(css='.post-preview .edit-button').present